bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS3Handler.java

```diff
@@ -0,0 +1,189 @@
+            Socket socket = channel.socket();
+            socket.setSoTimeout(SOCKET_TIMEOUT_SECONDS * 1000);
+            channel.configureBlocking(true);
+            channel.connect(new InetSocketAddress(config.ipAddress, DEFAULT_PS3_WAKE_ON_LAN_PORT));
+        } catch (IOException e) {
+            String message = e.getMessage();
+            if (message.contains("refused")) {
+                updateState(CHANNEL_POWER, OnOffType.ON);
+                updateStatus(ThingStatus.ONLINE);
+            } else if (message.contains("timed out") || message.contains("is down")) {
+                updateState(CHANNEL_POWER, OnOffType.OFF);
+            } else {
+                logger.debug("PS3 read power, IOException: {}", e.getMessage());
+            }
+        }
+    }
+
+    private void turnOnPS3() {
+        try {
+            InetAddress bcAddress = InetAddress.getByName("255.255.255.255");
+
+            // send WOL magic packet
+            byte[] magicPacket = makeWOLMagicPacket(thing.getProperties().get(Thing.PROPERTY_MAC_ADDRESS));
+            logger.debug("PS3 wol packet: {}", magicPacket);
+            DatagramPacket wakePacket = new DatagramPacket(magicPacket, magicPacket.length, bcAddress,
+                    DEFAULT_PS3_WAKE_ON_LAN_PORT);
+            // send discover
+            byte[] discover = "SRCH".getBytes(StandardCharsets.US_ASCII);
+            DatagramPacket srchPacket = new DatagramPacket(discover, discover.length, bcAddress,
+                    DEFAULT_PS3_WAKE_ON_LAN_PORT);
+            logger.debug("Search message: '{}'", discover);
+
+            // wait for responses
+            byte[] rxbuf = new byte[256];
+            DatagramPacket receivePacket = new DatagramPacket(rxbuf, rxbuf.length);
+            scheduler.execute(() -> wakeMethod(srchPacket, receivePacket, wakePacket, 34));
+        } catch (IOException e) {
+            logger.debug("No PS3 device found. Diagnostic: {}", e.getMessage());
+        }
+    }
+
+    private void wakeMethod(DatagramPacket srchPacket, DatagramPacket receivePacket, DatagramPacket wakePacket,
+            int triesLeft) {
+        try (DatagramSocket searchSocket = new DatagramSocket(); DatagramSocket wakeSocket = new DatagramSocket();) {
+            wakeSocket.setBroadcast(true);
+            searchSocket.setBroadcast(true);
+            searchSocket.setSoTimeout(1000);
+
+            searchSocket.send(srchPacket);
+            try {
+                searchSocket.receive(receivePacket);
+                logger.debug("PS3 started?: '{}'", receivePacket);
+                return;
+            } catch (SocketTimeoutException e) {
+                // try again
+            }
+            wakeSocket.send(wakePacket);
+            if (triesLeft <= 0) {
+                logger.debug("PS3 not started!");
+            } else {
+                scheduler.execute(() -> wakeMethod(srchPacket, receivePacket, wakePacket, triesLeft - 1));
```

Comment:
You should check whether or not this handler has been disposed of before you try scheduling more things here.